python/dgl/distributed/partition.py

```diff
@@ -873,10 +901,10 @@ def get_homogeneous(g, balance_ntypes):
                                                                       local_edges)
                     else:
                         edge_feats[etype + '/' + name] = g.edges[etype].data[name]
-        # Some adjustment for heterogeneous graphs.
-        if not g.is_homogeneous:
-            part.ndata['orig_id'] = F.gather_row(sim_g.ndata[NID], part.ndata['orig_id'])
-            part.edata['orig_id'] = F.gather_row(sim_g.edata[EID], part.edata['orig_id'])
+        # delete `orig_id` from ndata/edata
+        if reshuffle:
+            del part.ndata['orig_id']
+            del part.edata['orig_id']
```

Comment:
Instead of just patching it at the end, could you take the chance to refactor the code too? For example, could we not generate/save this `'orig_id'` at the beginning?

Reply:
orig_nids/eids are required during update `part_g.ndata[NTYPE]/edata[ETYPE]`. so we have to generate. I think we could avoid attach such data into `part_g.n/edata`.

Reply:
seems `'orig_id'` is always attached if `reshuffle=True` as shown below.
https://github.com/dmlc/dgl/blob/303b150f8c6b6dbd6ea2b3ca724ded64ae14cb5f/python/dgl/partition.py#L140-L152

Reply:
'orig_id' is useful and user may rely on it directly. as a result, I think we'd better leave `orig_id` in `part_g.n/edata`.

Reply:
ok

Reply:
@Rhett-Ying Can I revert the change of these lines? 'orig_id' is really useful for me to debug a code. With this key, I can verify whether the node embedding in a subgraph is identical to the result in the original graph.

Reply:
@chwan1016 I'm afraid not. You need to change your test logic: fetch such info from the saved files.

Reply:
@Rhett-Ying Could you please be more specific? `orig_id` is deleted in this piece of code and it does not exist in any file. Thanks.

Reply:
Oh, great! I didn't notice the `return_mapping` param before. Thanks!

Reply:
@Rhett-Ying Can we replace `return_mapping` with another param like `store_mapping` (similar with `store_id` in `node_subgraph()`)? Storing the `orig_id` with other graph data together would make the file structure cleaner. Thanks!

Reply:
It's intended to store `orig_id` separately from graph data as we don't always need such info together with graph data when loading a graph for distributed train even we saved such info.